src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2923,10 +2927,13 @@ public void HandleIDebugBreakpointErrorEvent2(IDebugEngine2 pEngine, IDebugProce
                             };
 
                             // TODO: currently VSCode will ignore the error message from "breakpoint" event, the workaround is to log the error to output window
-                            string outputMsg = string.Format(CultureInfo.CurrentCulture, AD7Resources.Error_FunctionBreakpoint, ad7BPRequest.FunctionPosition.Name, errorMsg);
-                            m_logger.WriteLine(LoggingCategory.DebuggerError, outputMsg);
+                            outputMessage = string.Format(CultureInfo.CurrentCulture, AD7Resources.Error_FunctionBreakpoint, ad7BPRequest.FunctionPosition.Name, errorMsg);
                         }
 
+                        if (!string.IsNullOrEmpty(outputMessage))
```

Comment:
>outputMessage [](start = 50, length = 13)

You probably want to pay attention to severity here (and maybe we need to tweak which severity we use on line 357) -- if this is just because of a long bind, or because GDB couldn't find a record for the specified source file in the PDB, we wouldn't want to dump it to the console.

Reply:
I added an additional check for severity greater than `BPET_SEV_LOW`. Line 357 of this file doesn't have anything related to severity. Did you mean to point to another location?

Reply:
Sorry, I meant [main branch AD7PendingBreakpoint.cs line 357](https://github.com/microsoft/MIEngine/blob/8456c49b4731971b885c2909f9feaac4bdcd494d/src/MIDebugEngine/AD7.Impl/AD7PendingBreakpoint.cs#L357). It looks like that uses BPET_GENERAL_WARNING.

I think what you probably want is to check if this is an error rather than a warning: `((errorType & enum_BP_ERROR_TYPE.BPET_TYPE_ERROR) != 0)`

---
In reply to: [602621170](https://github.com/microsoft/MIEngine/pull/1123#discussion_r602621170) [](ancestors = 602621170)

Reply:
I changed the error to `BPET_GENERAL_ERROR` and swapped in your check for error type.